Hexid

montjoie opened the pull request:
This PR fix the problem of malformated udev lines via two methods:
- prevent the use of string in idvendor/idproduct
- print the ID via "%04x"

This PR is based on top of enablekvm PR
This PR support also python3 for lavalab-gen.py

montjoie committed:
Remove hardcoded devices for lava-slave

The node devices was hardcoded for lava-slave.
For example, lava-master could never have devices.
Remove that hard-coding and made devices dynamic for any labname.

montjoie committed:
Fix the lab-slave-0 name

The boards.yaml documentation state that lava-slave should be named lab-slave-XXX
but docker-compose.template does not use this convention.

This force the use of this convention.

montjoie committed:
Remove old devices node

The "devices" node is now automaticly generated and added, so remove
them from docker-compose.template.

montjoie committed:
Reintroduce qemu KVM support

Since lava-docker must run on all arch, and that KVM is x86 specific,
KVM support was disabled for qemu.

Reintroduce it now as an option for each lab via the
host_has_cpuflag_kvm flag.

montjoie committed:
Document host_has_cpuflag_kvm

Now host_has_cpuflag_kvm is used, we need to document it.

montjoie committed:
Made lavalab-gen.py python3 compatible

lavalab-gen.py cannot be used with python3.
This patch made it compatible with both python2 and python3.

khilman commented on README.md:
```diff
@@ -91,7 +91,7 @@ So you have now:
       uart:
         idvendor: "0403"
-	idproduct: 6001
+	idproduct: "6001"
```
No, please do not require quotes.  Instead requre (and document in the examples) to use a hex prefix 0x.

khilman commented on boards.yaml:
```diff
@@ -1,4 +1,66 @@
+        hard_reset_command: /usr/local/bin/acme-cli -s 192.168.66.2 reset 1
+        power_off_command: /usr/local/bin/acme-cli -s 192.168.66.2 switch_off 1
+        power_on_command: /usr/local/bin/acme-cli -s 192.168.66.2 switch_on 1
+      uart:
```
The adding of all these boards should not be part of the master branch.  This should be in a lab-specific branch someplace.

montjoie replied:
This should never be in this PR, I have commited this by mistake sorry.

khilman commented on lavalab-gen.py:
```diff
@@ -92,6 +92,10 @@ def main(args):
+                # Prevent that convertion of the hex value
+                if type(idproduct) == int or type(idvendor) == int:
+                    print("Please encapsulate IDs for %s with quotes" % board_name)
+                    sys.exit(1)
```
Instead, read the values as hex (which will valditate they are hex) and then print them in udev rules as hex with `%04x`

montjoie replied:
print as hex is done, but we cannot read as hex, only as int so we cannot know if the value was with prefixed wit 0x or not

montjoie committed:
Force the use of hexadecimal values for product/vendor IDs

Using "idproduct: 6001" let python convert 6001 to 24577 and so corrupt
the UDEV line.
